### DISCUSSION TITLE
Reset All Atoms Easily

### DISCUSSION THREAD

**Discussion by johnsonav1992** [pmndrs/jotai#2120] [Q&A] [answered]
Hey there -

I'm sure you have gotten this question before, but in case you have not: is there an easy way to reset all atoms in my application at once without having to use useResetAtom for every one and call them all in-line? I am not using a provider, merely just a separate file with all of my atoms stored separately and then accessing them in my components. 

If this is not currently supported, will there be some support for this at any point in the future?

Thank you!

**Comment by dai-shi** [comment, marked as answer]
It may sound surprising, but from the library perspective, there's no notion of "all atoms". It doesn't even know how many atoms are stored. It's because the implementation is based on `WeakMap` and it's by design.

What you could do is to create a new store and replace with the old one. An easy way is to remount the `<Provider>` component if you use one.
Otherwise, you would need to reset all atoms by yourself.
There might be some other workarounds. Searching old discussions might or might not help...

**Reply by johnsonav1992** [reply]
Thanks, Daishi! I figured this was the case but I wanted to be 100% sure. I'm pretty new to Jotai so I wanted to have a clear picture and make sure I am using best practices. Have loved using it so far!

**Reply by alexrequelme** [reply]
Hi, I currently use a Provider to clean up specific atoms upon unmounting, like removing a songAtom when leaving a SongListPage. However, I'm facing an issue where, upon navigating to the SongListPage, the userAtom now resets to its initial value. This atom is defined globally and I want to keep the value.

**Reply by dai-shi** [reply]
You would need two stores. We are not sure about best practice, but technically you can do `useAtom(atomToBeKeptGlobally, { store: getDefaultStore() })`.

> This atom is defined globally

Please note that the atom is just a definition and if you used it under `<Provider>`, its value is not global (even if the atom definition is global).

**Reply by alexrequelme** [reply]
This code works, but im not sure if its right
```js
function SongListPage () {
  const defaultStore = getDefaultStore();

  return (
    <Provider store={defaultStore}>
     {/* Components */} 
    </Provider>
}
```

**Reply by alexrequelme** [reply]
Concerning the userAtom, I'm utilizing it in a provider-less mode, and I want only the songsAtom value to change according to the subtree.

**Reply by dai-shi** [reply]
The default store is the one used for the provider-less mode, so it doesn't make sense to pass it to `<Provider>`. It should be a new store.

**Reply by alexrequelme** [reply]
Is this way the preferable one, right?
```js
const user = useAtomValue(userAtom, { store: getDefaultStore() });
```
Indeed, I've been using the Provider primarily to clean up the songAtom upon page unmount, as it's specific to that page. However, the problem was that the userAtom value wasn't maintaining its global value.

**Reply by dai-shi** [reply]
I wouldn't prefer it, but it works. If it's just a few atoms, it should be fine.
My hesitation is explicitly using `store` doesn't seem ideal for pure atom model.
For complex use cases, something like `jotai-molecules` may model things better.

**Reply by alexrequelme** [reply]
Well, actually, it's the only 'global' atom I need on that page. I had to use the Provider because I noticed that when navigating back and then returning to the songs page, the values were still retained. Using the Provider is the only way to 'free up memory,' right?

**Reply by dai-shi** [reply]
> Using the Provider is the only way to 'free up memory,' right?

Yes. (Another way is to delete all atom references.)

**Comment by v1noid** [comment]
You can use [jotai-resetter](https://www.npmjs.com/package/jotai-resetter)

**Reply by johnsonav1992** [reply]
@sirarifarid Nice, thanks!
